src/genericTensorOps.hpp

```diff
@@ -262,19 +262,19 @@ constexpr std::ptrdiff_t SYM_SIZE = ( M * ( M + 1 ) ) / 2;
 
 /**
  * @return Return the largest absolute entry of @p vector.
- * @tparam M The size of @p vector.
+ * @tparam ISIZE The size of @p vector.
  * @tparam VECTOR The type of @p vector.
  * @param vector The vector to get the maximum entry of, of length M.
  */
-template< std::ptrdiff_t M, typename VECTOR >
+template< std::ptrdiff_t ISIZE, typename VECTOR >
 LVARRAY_HOST_DEVICE CONSTEXPR_WITHOUT_BOUNDS_CHECK inline
 auto maxAbsoluteEntry( VECTOR && vector )
 {
-  static_assert( M > 0, "M must be greater than zero." );
-  internal::checkSizes< M >( vector );
+  static_assert( ISIZE > 0, "M must be greater than zero." );
```

Comment:
A lot of these static assert messages keep referencing old parameter names. Would be best to keep them consistent... hopefully it's as simple as `s/M must/ISIZE must/g`, etc.

Also there are still some references in doxygen comment blocks, e.g. see `@param vector` line above. I would actually recommend removing duplication there, i.e. `@tparam ISIZE` line already explains the size of `vector`, no need to type it twice. This one is more annoying as it may no automate well.